File: javascript/resistor-color-duo/resistor-color-duo.js
const colors = { black: 0 , brown: 1 , red: 2 , orange: 3 , yellow: 4,  green: 5, blue: 6, violet: 7, grey: 8, white: 9 }

const value = colorKeys => {
  const ResistanceValue = colorKeys.map(colorKey => colors[colorKey]).join('');
  return Number(ResistanceValue);
}

export { value }

File: ruby/resistor-color-duo/resistor_color_duo.rb
class ResistorColorDuo
  COLORS = ['black', 'brown', 'red', 'orange', 'yellow', 'green', 'blue', 'violet', 'grey', 'white'].freeze

  def self.value(array)
    array.first(2).map { |color| COLORS.find_index(color) }.join.to_i
  end
end

